runtime/ftplugin/sh.vim

```diff
@@ -44,19 +44,30 @@ if (has("gui_win32") || has("gui_gtk")) && !exists("b:browsefilter")
   let b:undo_ftplugin ..= " | unlet! b:browsefilter"
 endif
 
-if get(b:, "is_bash", 0)
+let s:is_sh = get(b:, "is_sh", 0)
+let s:is_bash = get(b:, "is_bash", 0)
+let s:is_kornshell = get(b:, "is_kornshell", get(g:, "is_kornsh", 0))
```

Comment:
Do the introduced `s:` variables need to stay defined after  
the sourcing?

In `syntax/sh.vim`, there are `is_kornshell` queries and aren't  
`is_kornsh` ones.  Look for `is_kornshell` with both `get`s?

In `syntax/sh.vim`, `g:is_sh` and `g:is_bash` are recognised.  
Look for these too?